### DISCUSSION TITLE
phased-out indicator/frame alpha

### DISCUSSION THREAD

**Discussion by spamwax** [rgd87/Aptechka#79] [Q&A]
During last phase of denatrius, raid is divided into two groups that can't interact/target each other.
I can see the circle on frames to indicate the phase but the alpha of the frame is full as if I can target them.
This doesn't happen when, say, someone is in war mode and I see both circle marker as well as different alpha/color for them.

How do you suggest I work around this? can I use the debuff id that other half of raids gets to somehow adjust the alpha/color of the hp frames?

**Comment by d87** [comment]
Well in the past phased units were also always out of range. But if that's how it is now then i guess i'll have to add phase check along with range. Try https://github.com/rgd87/Aptechka/releases/tag/9.1-beta2

**Reply by spamwax** [reply]
@d87 I noticed the same issue happens with one of the adds in the same fight.
She basically has a "smoke bomb" effect that only lets you target/cast on players within the aura.
Aptechka however shows all units as in range and interact-able even if they are not in the smoke bomb.
It's called [Gloom Veil](https://www.wowhead.com/spell=340687)

**Reply by d87** [reply]
Well that's how the game shows it, If it's not an actual phase then that's a job for debuff highlighting probably, because you can step into the veil/bomb theoretically.

**Reply by spamwax** [reply]
I see. Was watching a stream in which the bars faded out when this happened.
I am gonna try to assign that debuff to inner_glow or something, do you have a suggesution to make it affect the alpha/fade of frames?

**Reply by d87** [reply]
Alright, made a widget for frame alpha and an additional debuff alert status.
https://github.com/rgd87/Aptechka/releases/tag/9.1-beta5

**Reply by spamwax** [reply]
Awesome 💯 , testing it now. will report back if any issue is encountered.

**Reply by spamwax** [reply]
ok, there is a bug in beta5 which happens when someone joins my group and they are not close so it shows them out of range.
but when they come close or zone into instance, they sill show as out of range/low alpha. I can however interact with them. I think this even happened without me using the new widget and status.
(/reload fixes it)
It even happened after i went afk, came back and my group had moved away from me but they were still shown as "in range"

related to my original request, i could assign the Gloom Veil and see the frames having different alpha. But I guess I wasn't clear originally (and this goes back to your original point you mentioned), I wanted to see who's in the same "phase" as me & if I could heal them, then they should have full alpha, if they have Gloom Veil & I don't (or vice vera) then I wanted the frame to have lower alpha, so basically a smoke bomb.
But it seems as you said, since this particular debuff doesn't trigger a real smoke bomb, this won't be possible.

Thanks for quick update and implementation though.

**Reply by d87** [reply]
So was it working most of the time or always getting randomly stuck?

https://github.com/rgd87/Aptechka/releases/tag/9.1-beta6
I hope it's fixed there. At least I tried running a couple matchmade dungeons and couldn't break it.

And another thing in beta6 is the debuff tooltips. Now if you hold Ctrl-Shift while hovering it'll show tooltips for all the debuffs.
There's also a keybind alternative for this

**Reply by d87** [reply]
Oh yeah, the Gloom Veil thing... Idk what magic addon you've seen but something like this requires basically a custom logic exclusively for it. Real Smoke Bomb is not displayed as anything too, it's just a debuff. what's even more annoying you can't distinguish between friendly & enemy smoke bombs without some specific checks. Anyway to phase people like that you'd need like a plugin specifically for denathrius. And maybe that's what it was, a weakaura or some crazy integration with boss mods

**Reply by d87** [reply]
Oh, I think I found the bug from beta5. It wasn't the range that was broken, but phasing. Maybe you don't have lua errors enabled, so you didn't see it?
https://github.com/rgd87/Aptechka/releases/tag/9.1-beta7

**Reply by spamwax** [reply]
I will try this today, I had buggrabber but didn't pay attention to it.

**Reply by spamwax** [reply]
it seems the issue hasn't been fixed. 
Invited a friend to group who was in maw, zoned into maw and the UI show in-range alpha 
![image](https://user-images.githubusercontent.com/1251233/123358585-ae5b4000-d539-11eb-893c-3e3b31f2227c.png)
BugSack didn't report any errors!
/reload fixes it though

**Reply by d87** [reply]
Damn, alright, I'll fix it in a bad way. Can't really deal with this while unsubbed

**Reply by d87** [reply]
https://github.com/rgd87/Aptechka/releases/tag/9.1-beta8

**Reply by d87** [reply]
Hey, if you'll make highlights for the new raid/dungeon, could you post them?

**Reply by spamwax** [reply]
I am using the [Glow WA](https://wago.io/SanctumGlows) that does it on frames.

**Reply by spamwax** [reply]
It has a lot of them but I can get you the spell ids of dangerous ones if you are no subbed

**Reply by d87** [reply]
just the spell names are fine of the ones that require attention/reaction, i'll figure out the rest. But ids are good too

**Reply by spamwax** [reply]
These are the ones for heavy dots or require healing:
358610
350482
350388
350217
350851
355506
352394
352385
357686
347670
358433

 multiple stacks of a dot that's not dangerous with just one stack:
352392
350073
352833
353398
353931
347807

dispellable/need mechanic attention
342077
350713
350039
350469
348759
351117
353929
353929
Banshee's Fury

**Reply by d87** [reply]
Thanks
